store/generated_key/eigenda/eigenda.go

```diff
@@ -71,7 +75,28 @@ func (e Store) Put(ctx context.Context, value []byte) ([]byte, error) {
 	}
 
 	dispersalStart := time.Now()
-	blobInfo, err := e.client.PutBlob(ctx, value)
+
+	// We attempt to disperse the blob to EigenDA up to 3 times, unless we get a 400 error on any attempt.
+	blobInfo, err := retry.DoWithData(
+		func() (*disperser.BlobInfo, error) {
+			return e.client.PutBlob(ctx, value)
+		},
+		retry.RetryIf(func(err error) bool {
+			var errAPI api.ErrorAPI
+			if errors.As(err, &errAPI) {
+				if errAPI.ErrorFault() == api.ErrorFaultClient {
+					// we don't retry 400 errors because there is no point,
+					// we are passing invalid data
+					return false
+				}
+			}
+			return true
+		}),
+		// only return the last error. If it is a 503, then the handler will convert it to an http 503
+		// to signify to the client (batcher) to failover to ethda b/c eigenda is temporarily down.
+		retry.LastErrorOnly(true),
```

Comment:
Last error is good. The only unhandled case is when the blob is confirmed, but ethereum failed to finalize. We shouldn't attribute it to EigenDA.

Reply:
nvm, after saw the PR 828, it is passthrough.